apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -200,16 +211,15 @@ public Builder serverUrl(@Nonnull String serverUrl) {
     }
 
     /**
-     * Set the configuration to be used for request/response http cache.
+     * Set the configuration to be used for request/response http cache. <em>Note:</em> you should supply an
+     * interceptor backed by this cache in your http client as well.
      *
-     * @param cacheStore       The store to use for reading and writing cached response.
-     * @param evictionStrategy EvictionStrategy decides when the data in the cacheStore becomes stale and is no longer
-     *                         valid
+     * @param httpCache The HttpCache to use.
      * @return The {@link Builder} object to be used for chaining method calls
+     * @see HttpCache#interceptor()
      */
-    public Builder httpCache(@Nonnull ResponseCacheStore cacheStore, @Nonnull EvictionStrategy evictionStrategy) {
-      this.httpCacheStore = checkNotNull(cacheStore, "cacheStore == null");
-      this.httpEvictionStrategy = checkNotNull(evictionStrategy, "evictionStrategy == null");
+    public Builder httpCache(@Nonnull HttpCache httpCache) {
```

Comment:
Another issue is with the constructor:
```
HttpCache(@Nonnull ResponseCacheStore cacheStore, @Nonnull EvictionStrategy evictionStrategy,
      @Nonnull ApolloLogger logger)
```
With 3rd argument, logger. So how user will instantiate this class, how logger will be provided?
Maybe instead of argument to constructor make it via setter